src/cli/playbook/ipa.rs

```diff
@@ -158,3 +158,94 @@ where
         breakdowns,
     }
 }
+
+pub async fn playbook_oprf_ipa<F>(
+    mut records: Vec<TestRawDataRecord>,
+    clients: &[MpcHelperClient; 3],
+    query_id: QueryId,
+    query_config: IpaQueryConfig,
+) -> IpaQueryResult
+where
+    F: PrimeField,
+    AdditiveShare<F>: Serializable,
+{
+    let mut buffers: [_; 3] = std::array::from_fn(|_| Vec::new());
+    let query_size = records.len();
+
+    let sz = <OprfReport<Timestamp, BreakdownKey, TriggerValue> as Serializable>::Size::USIZE;
+    for buffer in &mut buffers {
+        buffer.resize(query_size * sz, 0u8);
+    }
+
+    //TODO(richaj) This manual sorting will be removed once we have the PRF sharding in place
+    records.sort_by(|a, b| b.user_id.cmp(&a.user_id));
+
+    let shares: [Vec<OprfReport<Timestamp, BreakdownKey, TriggerValue>>; 3] =
+        records.iter().cloned().share();
+    zip(&mut buffers, shares).for_each(|(buf, shares)| {
+        for (share, chunk) in zip(shares, buf.chunks_mut(sz)) {
+            share.serialize(GenericArray::from_mut_slice(chunk));
+        }
+    });
+
+    let inputs = buffers.map(BodyStream::from);
+    tracing::info!("Starting query for OPRF");
```

Comment:
still needed?

Reply:
Yes, it is good to know that the query has started to run